### PR TITLE
fix: accounts name in send flow

### PR DESCRIPTION
**Pull request by gantunesr**
## **Description**

This change corrects the accounts name displayed incorrectly in the send flow list.

## **Related issues**

Fixes: X

## **Manual testing steps**

1. Create different types of accounts like hardware wallet and snap accounts
2. Start a send transaction
3. Notice the name of the accounts are displayed correctly

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by gantunesr** [comment]
@metamaskbot update-policies

**Comment by gantunesr** [comment]
@metamaskbot update-policies

**Review by owencraston — APPROVED** [review]

File: ui/pages/connected-accounts/connected-accounts.container.js [hunk 1]
```diff
@@ -21,6 +22,24 @@ const mapStateToProps = (state) => {
   const { activeTab } = state;
   const accountToConnect = getAccountToConnectToActiveTab(state);
   const connectedAccounts = getOrderedConnectedAccountsForActiveTab(state);
+  const internalAccounts = getInternalAccounts(state);
+  // Temporary fix until https://github.com/MetaMask/metamask-extension/pull/21553
+  const connectedAccountsWithName = connectedAccounts.map((account) => {
+    const updatedAccount = {
+      ...account,
+      name: internalAccounts.find(
```
Comment (k-g-j) [inline ui/pages/connected-accounts/connected-accounts.container.js:30]: Simular slight note about optimization here as above
```typescript
const internalAccountsMap = new Map(
    internalAccounts.map((acc) => [acc.address, acc]),
  );
const connectedAccountsWithName = connectedAccounts.map((account) => {
     const updatedAccount = {
       ...account,
       name: internalAccountsMap.get(account.address)?.metadata.name,
     };
     return updatedAccount;
   });

**Review by k-g-j — APPROVED** [review]
I do not have a hardware wallet to verify but the UI changes are as expected when manually testing using snap accounts.

Screenshot 2024-01-11 at 7 06 51 PM

File: ui/components/app/alerts/unconnected-account-alert/unconnected-account-alert.js [hunk 2]
```diff
@@ -33,8 +34,24 @@ const UnconnectedAccountAlert = () => {
   const connectedAccounts = useSelector(
     getOrderedConnectedAccountsForActiveTab,
   );
+  const internalAccounts = useSelector(getInternalAccounts);
+  // Temporary fix until https://github.com/MetaMask/metamask-extension/pull/21553
+  const connectedAccountsWithName = connectedAccounts.map((account) => {
+    account.name = internalAccounts.find(
```
[the remark this answers is not in the document]
Comment (gantunesr) [reply ui/components/app/alerts/unconnected-account-alert/unconnected-account-alert.js:40]: This is a good improvement but unless we're talking about thousands of elements this is most likely negligible

File: ui/pages/connected-accounts/connected-accounts.container.js (hunk 1 above)
Comment (gantunesr) [reply ui/pages/connected-accounts/connected-accounts.container.js:30]: Same as the other comment

**Pull request merged**